### PR TITLE
Live on Firebase

### PR DESCRIPTION
**Pull request by kenneth-blanton**

### CODE REVIEW

**Review by kenneth-blanton — COMMENTED** [review]
Currently not done with get and pull requests but all of the info from firebase hosting is in here.

Live @ https://teamwombatadmin1.web.app/

**Comment by joshuaow91** [comment]
might want to link up with @Srasulova she's also been working on the GET and PUT requests for the admin dashboard

File: wombat-admin/src/app/models/models.tsx [hunk 1]
```diff
@@ -18,13 +19,30 @@ type UserVisit = {
...
+  accessGivenBy: number; // admin id
 };
 
 
 type Admin = {
   id: number;
   username: string;
 };
+
+// type requestState = "AUTOMATED" | "MANUAL_OVERRIDE_REQUESTED" | "MANUAL_OVERRIDE_ACTIONED" | "MANUAL_OVERRIDE_TIMEOUT"
```
Comment (allaboutmike) [inline wombat-admin/src/app/models/models.tsx:33]: If these are not needed please remove them.

File: wombat-admin/src/app/page.tsx [hunk 2]
```diff
@@ -1,32 +1,51 @@
...
-  const [viewImageModal, setViewImageModal] = useState<Boolean>(false);
+  const [incomingDeniedRequests, setIncomingDeniedRequests] = useState<IncomingRequest[]>([]);
+  const [selectedRequest, setSelectedRequest] = useState<IncomingRequest | null>(null);
   const [imageSrc, setImageSrc] = useState<string | null>(null);
   const [activeTab, setActiveTab] = useState<string>('Daily Visits');
   const [currentPage, setCurrentPage] = useState(1);
 
+  const manualOverridenRequests = incomingDeniedRequests.filter(r => r.state === "MANUAL_OVERRIDE_REQUESTED");
+
+  useEffect(() => {
```
Comment (allaboutmike) [inline wombat-admin/src/app/page.tsx:26]: Can we put this in a hook?

File: wombat-admin/src/app/page.tsx [hunk 3]
```diff
@@ -1,32 +1,51 @@
...
+      }
+
+      const reqs: IncomingRequest[] = await res.json();
+      setIncomingDeniedRequests(reqs);
+    }
+
+    fetchData();
+
+    return () => setIncomingDeniedRequests([]);
+  }, []);
```
Comment (allaboutmike) [inline wombat-admin/src/app/page.tsx:42]: Needs the setIncomingDeniedRequests as a dependency.

File: wombat-admin/src/app/page.tsx [hunk 4]
```diff
@@ -44,34 +63,80 @@ export default function Home() {
     setActiveTab(tabName);
   };
 
-  // Custom hook for fetching requests data
-  const requests = useDataFetcher('http://localhost:4040/access_request');
+
   // Custom hook for fetching users data
-  const users = useDataFetcher('http://localhost:4040/users');
+  //const users = useDataFetcher('http://localhost:4040/users');
```
Comment (allaboutmike) [inline wombat-admin/src/app/page.tsx:68]: Nit please remove commented out code.

**Review by allaboutmike — CHANGES_REQUESTED** [review]
Overall I think we are ok, but a couple of things:

- do we need the .firebase files? These are generated. I'm not familiar with fire base, but if we don't need them on the repo let's remove / ignore them. 
- can we please clean out commented out source? If there is a need to add it back in we can put that in a Jira ticket.

File: wombat-admin/src/app/page.tsx [hunk 5]
```diff
@@ -78,32 +78,51 @@ export default function Home() {
...
     }
 
-    const requestOptions = {
-      method: 'PUT',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({ state: 'MANUAL_OVERRIDE_ACTIONED', date: "2024-04-17-05:00" }),
-    };
+    const reqsBody = { state: 'MANUAL_OVERRIDE_ACTIONED', approvalStatus: "DENIED", date: request.date };
```
Comment (joshuaow91) [inline wombat-admin/src/app/page.tsx:107]: if this is the approve request hook it should have an approvalStatus of "APPROVED"

**Review by FarhanDS-92 — APPROVED** [review]
LGTM!!

**Review by allaboutmike — APPROVED** [review]
Approving to unblock.

**Pull request merged**